packages/framework/src/Framework/Services/BuildTaskService.php

```diff
@@ -4,89 +4,114 @@
 
 namespace Hyde\Framework\Services;
 
+use Hyde\Facades\Config;
+use Hyde\Facades\Filesystem;
 use Hyde\Framework\Features\BuildTasks\BuildTask;
-use Hyde\Framework\Features\BuildTasks\PostBuildTasks\GenerateBuildManifest;
-use Hyde\Hyde;
+use Hyde\Framework\Features\BuildTasks\Contracts\RunsAfterBuild;
+use Hyde\Framework\Features\BuildTasks\Contracts\RunsBeforeBuild;
 use Illuminate\Console\OutputStyle;
+use Illuminate\Support\Str;
+use InvalidArgumentException;
+use function array_values;
+use function class_basename;
+use function is_bool;
+use function is_subclass_of;
+use function str_replace;
 
 /**
- * This service manages the build tasks that are called after the site has been compiled using the build command.
+ * This service manages the build tasks that are called before and after the site has been compiled using the build command.
+ *
+ * It is registered as a singleton in the Laravel service container.
  *
  * @see \Hyde\Framework\Testing\Feature\Services\BuildTaskServiceTest
+ * @see \Hyde\Framework\Testing\Unit\BuildTaskServiceUnitTest
  */
 class BuildTaskService
 {
-    /**
-     * Information for package developers: This offers a hook for packages to add custom build tasks.
-     * Make sure to add the fully qualified class name to the array and doing so by merging the array, not overwriting it.
-     *
-     * @var array<class-string<\Hyde\Framework\Features\BuildTasks\BuildTask>>
-     */
-    public static array $postBuildTasks = [];
+    /** @var array<string, class-string<\Hyde\Framework\Features\BuildTasks\BuildTask>> */
+    protected array $buildTasks = [];
```

Comment:
Refactor to store instantiated objects instead of class-strings

This makes it so much easier to mock them, it also provides native type support